web/frontend/src/components/slam.vue

```diff
@@ -177,83 +234,85 @@ const updateSLAM2dRefreshFrequency = async (name: string, time: 'manual' | 'off'
   }
 };
 
-const updateSLAM3dRefreshFrequency = async (name: string, time: 'manual' | 'off' | string) => {
-  refresh3DCancelled = true;
-  window.clearTimeout(slam3dTimeoutId);
-
-  if (time === 'manual') {
-    try {
-      const res = await fetchSLAMMap(name);
-      handleRefresh3dResponse(res);
-    } catch (error) {
-      handleError('fetchSLAMMap', error);
-    }
-  } else if (time === 'off') {
-    // do nothing
-  } else {
-    refresh3DCancelled = false;
-    scheduleRefresh3d(name, time);
-  }
-};
-
 const toggle2dExpand = () => {
   show2d = !show2d;
   updateSLAM2dRefreshFrequency(props.name, show2d ? selected2dValue : 'off');
 };
 
-const toggle3dExpand = () => {
-  show3d = !show3d;
-  updateSLAM3dRefreshFrequency(props.name, show3d ? selected3dValue : 'off');
-};
-
 const selectSLAM2dRefreshFrequency = () => {
   updateSLAM2dRefreshFrequency(props.name, selected2dValue);
 };
 
-const selectSLAMPCDRefreshFrequency = () => {
-  updateSLAM3dRefreshFrequency(props.name, selected3dValue);
-};
-
 const refresh2dMap = () => {
   updateSLAM2dRefreshFrequency(props.name, selected2dValue);
 };
 
-const refresh3dMap = () => {
-  updateSLAM3dRefreshFrequency(props.name, selected3dValue);
+const handle2dRenderClick = (event: THREE.Vector3) => {
+  updatedDest = true
+  threeJPos = event
+}
+
+const handleUpdateX = (event: CustomEvent<{ value: string }>) => {
+  threeJPos.x = Number.parseFloat(event.detail.value)
+  updatedDest = true
+}
+
+const handleUpdateY = (event: CustomEvent<{ value: string }>) => {
+  threeJPos.z = Number.parseFloat(event.detail.value)
+  updatedDest = true
+}
+
+const baseCopyPosition = () => {
+  const position: JSON = <JSON><unknown>{
+    "x": x,
+    "y": y,
+    "z": z,
+    "o_x": o_x,
+    "o_y": o_y,
+    "o_z": o_z,
+    "theta": theta
+  }
+  const asString = JSON.stringify(position)
+  copyToClipboardWithToast(asString)
 };
 
+const executeDelete = () => {
+  updatedDest = false
+  threeJPos = new THREE.Vector3()
+}
+
 </script>
 
 <template>
   <v-collapse
     :title="props.name"
     class="slam"
+    @toggle="toggle2dExpand()"
   >
     <v-breadcrumbs
       slot="title"
       crumbs="slam"
     />
-    <div class="h-auto border-x border-b border-black p-2">
-      <div class="container mx-auto">
-        <div class="pt-4">
-          <div class="flex items-center gap-2">
-            <v-switch
-              id="showImage"
-              :value="show2d ? 'on' : 'off'"
-              @input="toggle2dExpand()"
-            />
-            <span class="pr-2">View SLAM Map (2D)</span>
-          </div>
-          <div class="float-right pb-4">
+    <v-button
+        slot="header"
+        icon="stop-circle"
+        variant="danger"
+        label="STOP"
+        :disabled="moveClick ? 'true' : 'false'"
```

Comment:
this does not work as expected - when moveClick is false the button is not disabled.